indico/modules/events/abstracts/schemas.py

```diff
@@ -60,19 +60,19 @@ class Meta:
                   'ratings')
 
 
-class AbstractPersonLinkSchema(mm.ModelSchema):
+class AbstractPersonLinkSchema(mm.SQLAlchemyAutoSchema):
     class Meta:
         model = AbstractPersonLink
         fields = ('id', 'person_id', 'email', 'first_name', 'last_name', 'title', 'affiliation', 'address', 'phone',
                   'is_speaker', 'author_type')
 
 
-class AbstractSchema(mm.ModelSchema):
+class AbstractSchema(mm.SQLAlchemyAutoSchema):
     submitter = Nested(UserSchema)
     judge = Nested(UserSchema)
     modified_by = Nested(UserSchema)
-    duplicate_of = Nested('self', only=_basic_abstract_fields)
-    merged_into = Nested('self', only=_basic_abstract_fields)
+    duplicate_of = Nested(lambda: AbstractSchema, only=_basic_abstract_fields)
```

Comment:
I was just reading that `fields.Nested` can now take a callable (such as, `Nested(lambda: AbstractSchema(only=)`. But isn't this still the same thing?
```suggestion
    duplicate_of = Nested(AbstractSchema, only=_basic_abstract_fields)
```

Reply:
No, a class cannot reference itself during definition time. That's why they added support for passing a callable instead of the schema itself.

Reply:
Ah, I see, this is a particular case!